src/health_checker.cc

```diff
@@ -50,23 +51,18 @@ void HealthChecker::CleanupForTest() {
 std::set<std::string> HealthChecker::UnhealthyComponents() const {
   std::lock_guard<std::mutex> lock(mutex_);
   std::set<std::string> result(unhealthy_components_);
```

Comment:
Yeah, let me think about this one a little.

Reply:
This seems to me like a misuse of the store abstraction -- instead of just tracking the latest values, it would also maintain knowledge of connection status.

In any case, shouldn't these other unhealthy component checks go away when we re-establish the connection on drops?